include/cxxeffect/stream.hpp

```diff
@@ -39,7 +39,7 @@ namespace eff {
                 std::variant<R, std::pair<std::vector<O>, pull<F, O, R>>> v {r};
                 return v;
             };
-            return fr.template map<std::variant<R, std::pair<std::vector<O>, pull<F, O, R>>>>(wrapLeft);
+            return fr.map(wrapLeft);
```

Comment:
Is MSVC able to infer this type?

Reply:
Yes. The signature of map was changed so that you could pass it a lambda or other function directly, and it wraps it in std::funuction when creating the thunk.